src/plugins/data/public/data_sources/datasource/types.ts

```diff
@@ -3,49 +3,112 @@
  * SPDX-License-Identifier: Apache-2.0
  */
 
+import { DataSource } from './datasource';
+
 /**
  * @experimental These interfaces are experimental and might change in future releases.
  */
 
-import { IndexPatternsService } from '../../index_patterns';
-import { DataSourceType } from '../datasource_services';
-
 export interface IndexPatternOption {
   title: string;
   id: string;
 }
 
-export interface IDataSourceMetaData {
+export interface IDataSourceGroup {
   name: string;
 }
 
-export interface IDataSourceGroup {
-  name: string;
+export interface DataSetWithDataSource<T = unknown> {
+  ds: DataSource;
+  list: T[];
 }
 
-export interface ISourceDataSet {
-  ds: DataSourceType;
-  data_sets: Array<string | IndexPatternOption>;
+export interface IDataSetParams<T = {}> {
+  query: T;
 }
 
-// to-dos: add common interfaces for datasource
-// eslint-disable-next-line @typescript-eslint/no-empty-interface
-export interface IDataSetParams {}
+export interface IDataSourceQueryParams<T = {}> {
+  query: T;
+}
 
-// eslint-disable-next-line @typescript-eslint/no-empty-interface
-export interface IDataSourceQueryParams {}
+export interface IDataSourceQueryResult<T = {}> {
+  data: T;
+}
 
-// eslint-disable-next-line @typescript-eslint/no-empty-interface
-export interface IDataSourceQueryResult {}
+export enum ConnectionStatus {
+  Connected = 'connected',
+  Disconnected = 'disconnected',
+  Error = 'error',
+}
 
-export interface ConnectionStatus {
-  success: boolean;
-  info: string;
+export interface DataSourceConnectionStatus {
+  status: ConnectionStatus;
+  message: string;
+  error?: Error;
 }
 
-export interface DataSourceConfig {
-  name: string;
+export interface IDataSourceSettings<T extends IDataSourceMetadata = IDataSourceMetadata> {
+  id: string;
   type: string;
-  metadata: any;
-  indexPatterns: IndexPatternsService;
+  name: string;
+  metadata: T;
+}
+
+export interface IDataSourceMetadata {
+  ui: IDataSourceUISettings;
+}
+
+export interface IDataSourceUISelector {
+  displayDatasetsAsSource: boolean;
+}
+
+/**
+ * Represents the UI settings for a data source.
+ */
+export interface IDataSourceUISettings {
+  /**
+   * Controls UI elements related to data source selector.
+   */
+  selector: IDataSourceUISelector;
+
+  /**
+   * The display name of the data source.
+   */
+  label: string;
+
+  /**
+   * The group to which the data source belongs. This is used to group data sources in the selector.
+   */
+  groupType: DataSourceUIGroupType;
+
+  /**
+   * The display name of the data source type.
+   */
+  typeLabel: string;
+
+  /**
+   * A short description of the data source.
+   * @optional
+   */
+  description?: string;
+
+  /**
+   * URI of the icon representing the data source.
+   * @optional
+   */
+  icon?: string;
+}
+
+export interface IDataSourceDataSet<T = {}> {
+  dataSets: T;
+}
+
+export interface IDataSourceQueryResponse<T = {}> {
+  data: T;
+}
+
+export enum DataSourceUIGroupType {
```

Comment:
Same as above, do we have a plan on getting away from hardcoding datasource types in this service? These should be generic enough to allow different groups of datasources.